pkg/rbac/sharedData.go

```diff
@@ -51,36 +59,112 @@ var managedClusterResourceGvr = schema.GroupVersionResource{
 	Resource: "managedclusters",
 }
 
-func (cache *Cache) PopulateSharedCache(ctx context.Context) error {
+func (shared *SharedData) getPropertyTypes(cache *Cache, ctx context.Context) (map[string]string, error) {
+
+	// original query:
+	// select distinct key, jsonb_typeof(value) as datatype FROM search.resources,jsonb_each(data);
+	var selectDs *goqu.SelectDataset
+	//define schema:
+	schemaTable := goqu.S("search").Table("resources")
+	//data expression to get value and key
+	jsb := goqu.L("jsonb_each(?)", goqu.C("data"))
+	//select from these datasets
+	ds := goqu.From(schemaTable, jsb)
+	//select statement with orderby and distinct clause
+	selectDs = ds.Select(goqu.L("key"), goqu.L("jsonb_typeof(?)",
+		goqu.C("value")).As("datatype")).Distinct()
+
+	query, params, sqlerr := selectDs.ToSQL()
+	if sqlerr != nil {
+		klog.Errorf("Error building Search query: %s", sqlerr.Error())
+	}
+
+	klog.V(5).Infof("Query for property datatypes: [%s] ", query)
+	rows, err := cache.pool.Query(ctx, query, params...)
+	if err != nil {
+		klog.Errorf("Error resolving query [%s] with args [%+v]. Error: [%+v]", query, err)
+		return nil, err
+	}
+	defer rows.Close()
+
+	var key string
+	var value string
+	resourceTypeMap := make(map[string]string)
+	for rows.Next() {
+		err = rows.Scan(&key, &value)
+
+		if err != nil {
+			klog.Errorf("Error %s retrieving rows for query:%s", err.Error(), query)
+			continue
+		}
+		resourceTypeMap[key] = value
+
+	}
+	defer rows.Close()
+	//cache query:
+	shared.propTypes = PropTypesCache{resourceTypeMap}
+	shared.propTypeErr = err
+	shared.propTypeTime = time.Now()
+
+	return resourceTypeMap, err
+}
+
+func (cache *Cache) GetSharedData(ctx context.Context) (map[string]string, error) {
+	sharedData, err := cache.PopulateSharedCache(ctx)
+	if err != nil {
+		klog.Error("Error populating shared data cache: ", err)
+		return nil, err
+	}
+	//store only the PropTypeCache to use in outside of rbac module
+	SharedDataAccess := &PropTypesCache{
+		propType: sharedData.GetPropertyTypeCache(),
+	}
+	propTypesMap := SharedDataAccess.propType
+	return propTypesMap, nil
+}
+
+func (shared *SharedData) GetPropertyTypeCache() map[string]string {
+	return shared.propTypes.propType
+}
+
+func (cache *Cache) PopulateSharedCache(ctx context.Context) (*SharedData, error) {
 
 	if sharedCacheValid(&cache.shared) { //if all cache is valid we use cache data
 		klog.V(5).Info("Using shared data from cache.")
-		return nil
+		return &cache.shared, nil
 	} else { //get data and cache
 
 		var error error
+		_, err := cache.shared.getPropertyTypes(cache, ctx)
+		if cache.shared.propTypeErr == nil {
+			klog.V(6).Info("Successfully retrieved cluster scoped resources!")
+		} else {
+			error = err
```

Comment:
If there's more than 1 error in the logic below, only the last error will be returned.
I'd change this to log only if there's an error instead of logging on success. Logging errors is more helpful for debugging.

Reply:
That's a good point. Will log on error instead. This entire function needs to be changed so that these are running in parallel (in another PR)